cmd/skopeo/list_tags.go

```diff
@@ -113,23 +114,51 @@ func (opts *tagsOptions) run(args []string, stdout io.Writer) (retErr error) {
 		return fmt.Errorf("Invalid %q: does not specify a transport", args[0])
 	}
 
-	if transport.Name() != docker.Transport.Name() {
-		return fmt.Errorf("Unsupported transport '%v' for tag listing. Only '%v' currently supported", transport.Name(), docker.Transport.Name())
-	}
-
-	// Do transport-specific parsing and validation to get an image reference
-	imgRef, err := parseDockerRepositoryReference(args[0])
-	if err != nil {
-		return err
-	}
-
 	var repositoryName string
 	var tagListing []string
-	if err = retry.RetryIfNecessary(ctx, func() error {
-		repositoryName, tagListing, err = listDockerTags(ctx, sys, imgRef)
-		return err
-	}, opts.retryOpts); err != nil {
-		return err
+
+	switch transport.Name() {
+	case docker.Transport.Name():
+		// Do transport-specific parsing and validation to get an image reference
+		imgRef, err := parseDockerRepositoryReference(args[0])
+		if err != nil {
+			return err
+		}
+		if err = retry.RetryIfNecessary(ctx, func() error {
+			repositoryName, tagListing, err = listDockerTags(ctx, sys, imgRef)
+			return err
+		}, opts.retryOpts); err != nil {
+			return err
+		}
+	case archive.Transport.Name():
+		ref, err := archive.ParseReference(strings.TrimPrefix(args[0], archive.Transport.Name()+":"))
```

Comment:
Maybe do `alltransports.ParseImageName`, (along with an assertion that the transport didn’t change??) to minimize the amount of custom parsing code.

See also below about refusing references that contain more than a path.

Reply:
I'll update this part later when I have time

Reply:
> Maybe do `alltransports.ParseImageName`, (along with an assertion that the transport didn’t change??) to minimize the amount of custom parsing code.
> 
> See also below about refusing references that contain more than a path.

I'll update this part later when I have time